commit 31c7578ddb0fc15aa7247f2b8885956540031221
Author: Shiju Jose <shiju.jose@huawei.com>
Date:   Tue Feb 6 12:08:00 2024 +0000

    rasdaemon: ras-memory-failure-handler: update memory failure action page types
    
    Update memory failure action page types corresponding to the same in
    mm/memory-failure.c in the kernel.
    
    Signed-off-by: Shiju Jose <shiju.jose@huawei.com>
    Signed-off-by: Mauro Carvalho Chehab <mchehab@kernel.org>

diff --git a/ras-memory-failure-handler.c b/ras-memory-failure-handler.c
index 97e8840..a5acc08 100644
--- a/ras-memory-failure-handler.c
+++ b/ras-memory-failure-handler.c
@@ -26,10 +26,8 @@ enum mf_action_page_type {
 	MF_MSG_KERNEL_HIGH_ORDER,
 	MF_MSG_SLAB,
 	MF_MSG_DIFFERENT_COMPOUND,
-	MF_MSG_POISONED_HUGE,
 	MF_MSG_HUGE,
 	MF_MSG_FREE_HUGE,
-	MF_MSG_NON_PMD_HUGE,
 	MF_MSG_UNMAP_FAILED,
 	MF_MSG_DIRTY_SWAPCACHE,
 	MF_MSG_CLEAN_SWAPCACHE,
@@ -41,7 +39,6 @@ enum mf_action_page_type {
 	MF_MSG_CLEAN_LRU,
 	MF_MSG_TRUNCATED_LRU,
 	MF_MSG_BUDDY,
-	MF_MSG_BUDDY_2ND,
 	MF_MSG_DAX,
 	MF_MSG_UNSPLIT_THP,
 	MF_MSG_UNKNOWN,
@@ -64,10 +61,8 @@ static const struct {
 	{ MF_MSG_KERNEL_HIGH_ORDER, "high-order kernel page"},
 	{ MF_MSG_SLAB, "kernel slab page"},
 	{ MF_MSG_DIFFERENT_COMPOUND, "different compound page after locking"},
-	{ MF_MSG_POISONED_HUGE, "huge page already hardware poisoned"},
 	{ MF_MSG_HUGE, "huge page"},
 	{ MF_MSG_FREE_HUGE, "free huge page"},
-	{ MF_MSG_NON_PMD_HUGE, "non-pmd-sized huge page"},
 	{ MF_MSG_UNMAP_FAILED, "unmapping failed page"},
 	{ MF_MSG_DIRTY_SWAPCACHE, "dirty swapcache page"},
 	{ MF_MSG_CLEAN_SWAPCACHE, "clean swapcache page"},
@@ -79,7 +74,6 @@ static const struct {
 	{ MF_MSG_CLEAN_LRU, "clean LRU page"},
 	{ MF_MSG_TRUNCATED_LRU, "already truncated LRU page"},
 	{ MF_MSG_BUDDY, "free buddy page"},
-	{ MF_MSG_BUDDY_2ND, "free buddy page (2nd try)"},
 	{ MF_MSG_DAX, "dax page"},
 	{ MF_MSG_UNSPLIT_THP, "unsplit thp"},
 	{ MF_MSG_UNKNOWN, "unknown page"},
